src/WebformCivicrmPostProcess.php

```diff
@@ -675,7 +675,7 @@ private function findDuplicateContact($contact) {
         $m = $contact['address'][1]['master_id'];
         // If master address is exposed to the form, use it
         if (!empty($this->data['contact'][$m]['address'][1])) {
-          $contact['address'][1] = $this->data['contact']['address'][1];
+          $contact['address'][1] = $this->data['contact'][$m]['address'][1];
         }
         // Else look up the master contact's address
         elseif (!empty($this->existing_contacts[$m])) {
```

Comment:
@MegaphoneJon - -> looks like we could use this in D7 as well. Do you want to backport it?

https://github.com/colemanw/webform_civicrm/blob/7.x-5.x/includes/wf_crm_webform_postprocess.inc#L590-L596